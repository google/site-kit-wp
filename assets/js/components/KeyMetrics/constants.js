/**
 * Key Metrics components - constants.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const KEY_METRICS_SETUP_CTA_WIDGET_SLUG = 'key-metrics-setup-cta-widget';

export const KEY_METRICS_SELECTION_PANEL_OPENED_KEY =
	'googlesitekit-key-metrics-selection-panel-opened';
export const KEY_METRICS_SELECTION_FORM = 'key-metrics-selection-form';
export const KEY_METRICS_SELECTED = 'key-metrics-selected';
export const EFFECTIVE_SELECTION = 'key-metrics-effective-selection';
export const UNSTAGED_SELECTION = 'key-metrics-unstaged-selection';
export const MIN_SELECTED_METRICS_COUNT = 2;
export const MAX_SELECTED_METRICS_COUNT = 4;
export const MAX_SELECTED_METRICS_COUNT_WITH_CONVERSION_EVENTS = 8;
export const KEY_METRICS_GROUP_CURRENT = {
	SLUG: 'current-selection',
	LABEL: __( 'Current selection', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_SUGGESTED = {
	SLUG: 'suggested',
	LABEL: __( 'Suggested', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_VISITORS = {
	SLUG: 'visitors',
	LABEL: __( 'Visitors', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_DRIVING_TRAFFIC = {
	SLUG: 'driving-traffic',
	LABEL: __( 'Driving traffic', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_GENERATING_LEADS = {
	SLUG: 'generating-leads',
	LABEL: __( 'Generating leads', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_SELLING_PRODUCTS = {
	SLUG: 'selling-products',
	LABEL: __( 'Selling products', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_CONTENT_PERFORMANCE = {
	SLUG: 'content-performance',
	LABEL: __( 'Content performance', 'google-site-kit' ),
};
