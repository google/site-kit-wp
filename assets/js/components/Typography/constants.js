/**
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const TYPE_DISPLAY = 'display';
export const TYPE_HEADLINE = 'headline';
export const TYPE_TITLE = 'title';
export const TYPE_BODY = 'body';
export const TYPE_LABEL = 'label';

export const VALID_TYPES = [
	TYPE_BODY,
	TYPE_DISPLAY,
	TYPE_HEADLINE,
	TYPE_LABEL,
	TYPE_TITLE,
];

export const SIZE_SMALL = 'small';
export const SIZE_MEDIUM = 'medium';
export const SIZE_LARGE = 'large';

export const VALID_SIZES = [ SIZE_SMALL, SIZE_MEDIUM, SIZE_LARGE ];
