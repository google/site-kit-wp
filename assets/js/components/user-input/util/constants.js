/**
 * User Input Answers.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const USER_INPUT_QUESTIONS_PURPOSE = 'purpose';
export const USER_INPUT_QUESTION_POST_FREQUENCY = 'postFrequency';
export const USER_INPUT_QUESTIONS_GOALS = 'goals';

export const USER_INPUT_QUESTIONS_LIST = [
	USER_INPUT_QUESTIONS_PURPOSE,
	USER_INPUT_QUESTION_POST_FREQUENCY,
	USER_INPUT_QUESTIONS_GOALS,
];

export const USER_INPUT_MAX_ANSWERS = {
	[ USER_INPUT_QUESTIONS_PURPOSE ]: 1,
	[ USER_INPUT_QUESTION_POST_FREQUENCY ]: 1,
	[ USER_INPUT_QUESTIONS_GOALS ]: 3,
};

export const USER_INPUT_CURRENTLY_EDITING_KEY =
	'googlesitekit-user-input-currently-editing';

export const FORM_USER_INPUT_QUESTION_NUMBER = 'user_input_question_number';
export const FORM_USER_INPUT_QUESTION_SNAPSHOT = 'user_input_question_snapshot';
export const USER_INPUT_LEGACY_SITE_PURPOSE_DISMISSED_ITEM_KEY =
	'user-input-legacy-site-purpose-dismissed-item';

/**
 * Gets available questions for user input settings.
 *
 * @since 1.139.0
 * @private
 *
 * @return {Object} Questions object.
 */
export function getUserInputQuestions() {
	return [
		{
			title: __(
				'What is the main purpose of this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answer, Site Kit will tailor the metrics you see on your dashboard to help you track progress towards your specific goals',
				'google-site-kit'
			),
		},
		{
			title: __(
				'How often do you create new content for this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answer, Site Kit will suggest new features for your dashboard related to content creation',
				'google-site-kit'
			),
		},
		{
			title: __(
				'What are your top goals for this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answers, Site Kit will tailor the metrics and recommendations you see on your dashboard to help you make progress in these areas',
				'google-site-kit'
			),
		},
	];
}

/**
 * Gets available answers for user input settings.
 *
 * @since 1.20.0
 * @private
 *
 * @return {Object} Answers object.
 */
export function getUserInputAnswers() {
	return {
		USER_INPUT_ANSWERS_PURPOSE: {
			sell_products_or_service: __(
				'Sell products or services',
				'google-site-kit'
			),
			sell_products: __( 'Sell products', 'google-site-kit' ),
			provide_services: __( 'Provide services', 'google-site-kit' ),
			monetize_content: __(
				'Monetize content (with ads or affiliate links)',
				'google-site-kit'
			),
			publish_blog: __( 'Publish a blog', 'google-site-kit' ),
			publish_news: __( 'Publish news content', 'google-site-kit' ),
			share_portfolio: __(
				'Share a business card or portfolio to represent me or my company online',
				'google-site-kit'
			),
			other: __( 'Other', 'google-site-kit' ),
		},
		USER_INPUT_ANSWERS_POST_FREQUENCY: {
			never: __( 'Never', 'google-site-kit' ),
			daily: __( 'Daily', 'google-site-kit' ),
			weekly: __( 'Weekly', 'google-site-kit' ),
			monthly: __( 'Monthly', 'google-site-kit' ),
			other: __( 'Other', 'google-site-kit' ),
		},
		USER_INPUT_ANSWERS_GOALS: {
			retaining_visitors: __(
				'Retain visitors, turn them into loyal readers or customers',
				'google-site-kit'
			),
			improving_performance: __(
				'Improve speed and performance',
				'google-site-kit'
			),
			finding_new_topics: __(
				'Find new topics to write about that connect with my audience',
				'google-site-kit'
			),
			growing_audience: __( 'Grow my audience', 'google-site-kit' ),
			expanding_business: __(
				'Expand my business into new cities, states or markets',
				'google-site-kit'
			),
			generating_revenue: __(
				'Generate more revenue',
				'google-site-kit'
			),
			help_better_rank: __(
				'Help my content rank in a better position in Google search results',
				'google-site-kit'
			),
			understanding_content_performance: __(
				'Understand which content is performing best',
				'google-site-kit'
			),
			encourage_to_post: __(
				'Tips for generating and posting engaging content updates',
				'google-site-kit'
			),
			other: __( 'Other', 'google-site-kit' ),
		},
	};
}

/**
 * Gets available answer descriptions for user input settings.
 *
 * @since 1.139.0
 * @private
 *
 * @return {Object} Answer descriptions object.
 */
export function getUserInputAnswersDescription() {
	return {
		USER_INPUT_ANSWERS_PURPOSE: {
			sell_products_or_service: __(
				'E.g. selling products like devices, apparel, equipment, etc. or offering services like courses, consulting, tutoring, etc.',
				'google-site-kit'
			),
			sell_products: __(
				'E.g. selling devices, apparel, equipment, etc.',
				'google-site-kit'
			),
			provide_services: __(
				'E.g. offering courses, consulting, tutoring, etc.',
				'google-site-kit'
			),
			monetize_content: __(
				'Using display ads, affiliate links, sponsored content, etc.',
				'google-site-kit'
			),
			publish_blog: __(
				'Writing on a topic youâ€™re passionate about, no focus on monetizing content',
				'google-site-kit'
			),
			publish_news: __(
				'E.g. local news, investigative pieces, interviews, etc.',
				'google-site-kit'
			),
			share_portfolio: __(
				'My website represents me or my company',
				'google-site-kit'
			),
			other: undefined,
		},
	};
}
