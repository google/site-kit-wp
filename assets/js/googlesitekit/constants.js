/**
 * Core constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const VIEW_CONTEXT_MAIN_DASHBOARD = 'mainDashboard';
export const VIEW_CONTEXT_ENTITY_DASHBOARD = 'entityDashboard';
export const VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY = 'mainDashboardViewOnly';
export const VIEW_CONTEXT_ENTITY_DASHBOARD_VIEW_ONLY =
	'entityDashboardViewOnly';
export const VIEW_CONTEXT_POSTS_LIST = 'postsList';
export const VIEW_CONTEXT_USER_INPUT = 'userInput';
export const VIEW_CONTEXT_ACTIVATION = 'activation';
export const VIEW_CONTEXT_SPLASH = 'splash';
export const VIEW_CONTEXT_ADMIN_BAR = 'adminBar';
export const VIEW_CONTEXT_ADMIN_BAR_VIEW_ONLY = 'adminBarViewOnly';
export const VIEW_CONTEXT_SETTINGS = 'settings';
export const VIEW_CONTEXT_AD_BLOCKING_RECOVERY = 'adBlockingRecovery';
export const VIEW_CONTEXT_MODULE = 'module';
export const VIEW_CONTEXT_WP_DASHBOARD = 'wpDashboard';
export const VIEW_CONTEXT_WP_DASHBOARD_VIEW_ONLY = 'wpDashboardViewOnly';
export const VIEW_CONTEXT_MODULE_SETUP = 'moduleSetup';
export const VIEW_CONTEXT_METRIC_SELECTION = 'metricSelection';

// Anchor ID Constants
export const ANCHOR_ID_KEY_METRICS = 'key-metrics';
export const ANCHOR_ID_TRAFFIC = 'traffic';
export const ANCHOR_ID_CONTENT = 'content';
export const ANCHOR_ID_SPEED = 'speed';
export const ANCHOR_ID_MONETIZATION = 'monetization';

// Site Kit only view contexts
export const SITE_KIT_VIEW_CONTEXTS = [
	VIEW_CONTEXT_MAIN_DASHBOARD,
	VIEW_CONTEXT_ENTITY_DASHBOARD,
	VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_ENTITY_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_USER_INPUT,
	VIEW_CONTEXT_SPLASH,
	VIEW_CONTEXT_SETTINGS,
	VIEW_CONTEXT_MODULE_SETUP,
	VIEW_CONTEXT_METRIC_SELECTION,
];

// Site Kit view-only contexts
export const SITE_KIT_VIEW_ONLY_CONTEXTS = [
	VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_ENTITY_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_ADMIN_BAR_VIEW_ONLY,
	VIEW_CONTEXT_WP_DASHBOARD_VIEW_ONLY,
];
