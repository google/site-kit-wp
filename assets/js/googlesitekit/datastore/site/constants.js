/**
 * `core/site` data store: constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { MODULE_SLUG_ADS } from '@/js/modules/ads/constants';
import { MODULE_SLUG_ANALYTICS_4 } from '@/js/modules/analytics-4/constants';
import { MODULE_SLUG_TAGMANAGER } from '@/js/modules/tagmanager/constants';

export const CORE_SITE = 'core/site';

export const AMP_MODE_PRIMARY = 'primary';
export const AMP_MODE_SECONDARY = 'secondary';

export const GOOGLE_TAG_GATEWAY_MODULES = [
	MODULE_SLUG_ANALYTICS_4,
	MODULE_SLUG_ADS,
	MODULE_SLUG_TAGMANAGER,
];
