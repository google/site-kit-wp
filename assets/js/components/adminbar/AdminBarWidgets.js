/**
 * Admin Bar widgets.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { Fragment } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { useSelect } from 'googlesitekit-data';
import AdminBarImpressions from './AdminBarImpressions';
import AdminBarClicks from './AdminBarClicks';
import AdminBarUniqueVisitorsGA4 from './AdminBarUniqueVisitorsGA4';
import AdminBarSessionsGA4 from './AdminBarSessionsGA4';
import AdminBarActivateAnalyticsCTA from './AdminBarActivateAnalyticsCTA';
import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
import { Row, Cell } from '../../material-components';
import { withWidgetComponentProps } from '../../googlesitekit/widgets/util/get-widget-component-props';

// Widget slugs.
const WIDGET_IMPRESSIONS = 'adminBarImpressions';
const WIDGET_CLICKS = 'adminBarClicks';
const WIDGET_VISITORS = 'adminBarUniqueVisitors';
const WIDGET_SESSIONS = 'adminBarSessions';

// Search Console widgets.
const AdminBarImpressionsWidget =
	withWidgetComponentProps( WIDGET_IMPRESSIONS )( AdminBarImpressions );
const AdminBarClicksWidget =
	withWidgetComponentProps( WIDGET_CLICKS )( AdminBarClicks );

// Analytics 4 Widgets.
const AdminBarUniqueVisitorsGA4Widget = withWidgetComponentProps(
	WIDGET_VISITORS
)( AdminBarUniqueVisitorsGA4 );
const AdminBarSessionsGA4Widget =
	withWidgetComponentProps( WIDGET_SESSIONS )( AdminBarSessionsGA4 );

export default function AdminBarWidgets() {
	const analyticsModuleAvailable = useSelect( ( select ) =>
		select( CORE_MODULES ).isModuleAvailable( 'analytics-4' )
	);
	const analyticsModuleConnected = useSelect( ( select ) =>
		select( CORE_MODULES ).isModuleConnected( 'analytics-4' )
	);
	const analyticsModuleActive = useSelect( ( select ) =>
		select( CORE_MODULES ).isModuleActive( 'analytics-4' )
	);
	const canViewSharedAnalytics = useSelect( ( select ) =>
		select( CORE_USER ).hasAccessToShareableModule( 'analytics-4' )
	);
	const canViewSharedSearchConsole = useSelect( ( select ) =>
		select( CORE_USER ).hasAccessToShareableModule( 'search-console' )
	);

	const searchConsoleSize = canViewSharedAnalytics
		? { lg: 3, md: 2 }
		: { lg: 6, md: 4 };
	const analyticsSize = canViewSharedSearchConsole
		? { lg: 3, md: 2 }
		: { lg: 6, md: 4 };

	return (
		<Fragment>
			<Row>
				{ canViewSharedSearchConsole && (
					<Fragment>
						<Cell
							lgSize={ searchConsoleSize.lg }
							mdSize={ searchConsoleSize.md }
						>
							<AdminBarImpressionsWidget />
						</Cell>
						<Cell
							lgSize={ searchConsoleSize.lg }
							mdSize={ searchConsoleSize.md }
						>
							<AdminBarClicksWidget />
						</Cell>
					</Fragment>
				) }

				{ analyticsModuleConnected &&
					analyticsModuleActive &&
					canViewSharedAnalytics && (
						<Fragment>
							<Fragment>
								<Cell
									lgSize={ analyticsSize.lg }
									mdSize={ analyticsSize.md }
								>
									<AdminBarUniqueVisitorsGA4Widget />
								</Cell>
								<Cell
									lgSize={ analyticsSize.lg }
									mdSize={ analyticsSize.md }
								>
									<AdminBarSessionsGA4Widget />
								</Cell>
							</Fragment>
						</Fragment>
					) }

				{ analyticsModuleAvailable &&
					( ! analyticsModuleConnected ||
						! analyticsModuleActive ) && (
						<Cell lgSize={ 6 } mdSize={ 4 }>
							<AdminBarActivateAnalyticsCTA />
						</Cell>
					) }
			</Row>
		</Fragment>
	);
}
