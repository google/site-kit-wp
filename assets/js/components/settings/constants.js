/**
 * Settings constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { MODULE_SLUG_ADS } from '../../modules/ads/datastore/constants';
import { MODULE_SLUG_SIGN_IN_WITH_GOOGLE } from '../../modules/sign-in-with-google/datastore/constants';
import { MODULE_SLUG_READER_REVENUE_MANAGER } from '../../modules/reader-revenue-manager/datastore/constants';

export const NEW_MODULES = [
	MODULE_SLUG_ADS,
	MODULE_SLUG_READER_REVENUE_MANAGER,
	MODULE_SLUG_SIGN_IN_WITH_GOOGLE,
];

export const BETA_MODULES = [ MODULE_SLUG_SIGN_IN_WITH_GOOGLE ];

export const EXPERIMENTAL_MODULES = [];
