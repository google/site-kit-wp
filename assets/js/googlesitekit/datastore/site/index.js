/**
 * `core/site` data store
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { combineStores, commonStore } from 'googlesitekit-data';
import cache from './cache';
import connection from './connection';
import consentMode from './consent-mode';
import conversionTracking from './conversion-tracking';
import errors from './errors';
import firstPartyMode from './first-party-mode';
import html from './html';
import info from './info';
import reset from './reset';
import enableAutoUpdate from './enable-auto-update';
import settings from './settings';
import urls from './urls';
import developerPlugin from './developer-plugin';
import { CORE_SITE } from './constants';
import notifications from './notifications';
import { createErrorStore } from '../../data/create-error-store';

const store = combineStores(
	commonStore,
	connection,
	consentMode,
	conversionTracking,
	errors,
	firstPartyMode,
	html,
	info,
	developerPlugin,
	reset,
	enableAutoUpdate,
	settings,
	urls,
	notifications,
	cache,
	createErrorStore( CORE_SITE )
);

export const initialState = store.initialState;
export const actions = store.actions;
export const controls = store.controls;
export const reducer = store.reducer;
export const resolvers = store.resolvers;
export const selectors = store.selectors;

export const registerStore = ( registry ) => {
	registry.registerStore( CORE_SITE, store );
};

export default store;
