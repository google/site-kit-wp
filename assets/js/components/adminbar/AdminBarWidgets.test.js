/**
 * AdminBarWidgets component tests.
 *
 * Site Kit by Google, Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {
	render,
	createTestRegistry,
	provideModules,
	provideUserCapabilities,
	muteFetch,
	provideUserAuthentication,
} from '../../../../tests/js/test-utils';
import coreModulesFixture from '@/js/googlesitekit/modules/datastore/__fixtures__';
import { CORE_MODULES } from '@/js/googlesitekit/modules/datastore/constants';
import { CORE_SITE } from '@/js/googlesitekit/datastore/site/constants';
import { CORE_USER } from '@/js/googlesitekit/datastore/user/constants';
import { VIEW_CONTEXT_ADMIN_BAR_VIEW_ONLY } from '@/js/googlesitekit/constants';
import AdminBarWidgets from './AdminBarWidgets';
import { MODULE_SLUG_ANALYTICS_4 } from '@/js/modules/analytics-4/constants';

describe( 'AdminBarWidgets', () => {
	let registry;

	beforeEach( () => {
		registry = createTestRegistry();

		provideModules( registry );
		provideUserCapabilities( registry );

		registry.dispatch( CORE_USER ).receiveGetAuthentication( {
			authenticated: true,
			needsReauthentication: false,
		} );

		registry.dispatch( CORE_SITE ).receiveSiteInfo( {
			adminURL: 'http://example.com/wp-admin/',
		} );

		fetchMock.get(
			new RegExp(
				'/google-site-kit/v1/modules/search-console/data/searchanalytics'
			),
			{
				body: [
					{
						clicks: 123,
						ctr: 8.91,
						impressions: 4567,
						position: 23.456,
					},
				],
			}
		);

		muteFetch(
			new RegExp(
				'/google-site-kit/v1/modules/search-console/data/data-available'
			)
		);
	} );

	it( 'should render the Admin Bar Widgets, including the Activate Analytics CTA', async () => {
		const { container, getByText, waitForRegistry } = render(
			<AdminBarWidgets />,
			{
				registry,
			}
		);

		await waitForRegistry();

		expect( container ).toMatchSnapshot();

		expect( getByText( /Set up Google Analytics/ ) ).toBeInTheDocument();
	} );

	it( 'should not render the Activate Analytics CTA when the Analytics module is not available', async () => {
		registry
			.dispatch( CORE_MODULES )
			.receiveGetModules(
				coreModulesFixture.filter(
					( { slug } ) => slug !== MODULE_SLUG_ANALYTICS_4
				)
			);

		const { container, queryByText, waitForRegistry } = render(
			<AdminBarWidgets />,
			{
				registry,
			}
		);

		await waitForRegistry();

		expect( container ).toMatchSnapshot();

		expect(
			queryByText( /Set up Google Analytics/ )
		).not.toBeInTheDocument();
	} );

	it( 'should render the Admin Bar Widgets for the view only user if the module is shared', async () => {
		provideUserAuthentication( registry, { authenticated: false } );
		provideUserCapabilities( registry, {
			'googlesitekit_read_shared_module_data::["search-console"]': true,
		} );

		const { getByText, queryByText, waitForRegistry } = render(
			<AdminBarWidgets />,
			{
				registry,
				viewContext: VIEW_CONTEXT_ADMIN_BAR_VIEW_ONLY,
			}
		);

		await waitForRegistry();

		// Verify that the Search Console widgets are rendered.
		expect( getByText( /total impressions/i ) ).toBeInTheDocument();
		expect( getByText( /total clicks/i ) ).toBeInTheDocument();

		// Verify that the Analytics widgets are not rendered.
		expect( queryByText( /total users/i ) ).not.toBeInTheDocument();
		expect( queryByText( /total sessions/i ) ).not.toBeInTheDocument();
	} );

	it( 'should not render the Admin Bar Widgets for the view only user if the module is not shared', async () => {
		provideUserAuthentication( registry, { authenticated: false } );

		const { queryByText, waitForRegistry } = render( <AdminBarWidgets />, {
			registry,
			viewContext: VIEW_CONTEXT_ADMIN_BAR_VIEW_ONLY,
		} );

		await waitForRegistry();

		expect( queryByText( /total impressions/i ) ).not.toBeInTheDocument();
	} );
} );
