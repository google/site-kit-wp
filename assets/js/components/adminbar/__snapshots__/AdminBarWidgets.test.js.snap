// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminBarWidgets should not render the Activate Analytics CTA when the Analytics module is not available 1`] = `
<div>
  <div
    class="mdc-layout-grid__inner"
  >
    <div
      class="mdc-layout-grid__cell mdc-layout-grid__cell--span-6-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div
        class="googlesitekit-data-block overview-total-impressions googlesitekit-data-block--default"
        tabindex="-1"
      >
        <div
          class="googlesitekit-data-block__title-datapoint-wrapper"
        >
          <h3
            class=" googlesitekit-subheading-1 googlesitekit-data-block__title "
          >
            <span
              class="googlesitekit-data-block__title-inner"
            >
              Total Impressions
            </span>
          </h3>
          <div
            class="googlesitekit-data-block__datapoint"
          >
            4.6K
          </div>
        </div>
        <div
          class="googlesitekit-data-block__change-source-wrapper"
        >
          <div
            class="googlesitekit-data-block__change googlesitekit-data-block__change--no-change"
          >
            <span
              class="googlesitekit-data-block__value"
            >
              0%
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mdc-layout-grid__cell mdc-layout-grid__cell--span-6-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div
        class="googlesitekit-data-block overview-total-clicks googlesitekit-data-block--default"
        tabindex="-1"
      >
        <div
          class="googlesitekit-data-block__title-datapoint-wrapper"
        >
          <h3
            class=" googlesitekit-subheading-1 googlesitekit-data-block__title "
          >
            <span
              class="googlesitekit-data-block__title-inner"
            >
              Total Clicks
            </span>
          </h3>
          <div
            class="googlesitekit-data-block__datapoint"
          >
            123
          </div>
        </div>
        <div
          class="googlesitekit-data-block__change-source-wrapper"
        >
          <div
            class="googlesitekit-data-block__change googlesitekit-data-block__change--no-change"
          >
            <span
              class="googlesitekit-data-block__value"
            >
              0%
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AdminBarWidgets should render the Admin Bar Widgets, including the Activate Analytics CTA 1`] = `
<div>
  <div
    class="mdc-layout-grid__inner"
  >
    <div
      class="mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-2-tablet"
    >
      <div
        class="googlesitekit-data-block overview-total-impressions googlesitekit-data-block--default"
        tabindex="-1"
      >
        <div
          class="googlesitekit-data-block__title-datapoint-wrapper"
        >
          <h3
            class=" googlesitekit-subheading-1 googlesitekit-data-block__title "
          >
            <span
              class="googlesitekit-data-block__title-inner"
            >
              Total Impressions
            </span>
          </h3>
          <div
            class="googlesitekit-data-block__datapoint"
          >
            4.6K
          </div>
        </div>
        <div
          class="googlesitekit-data-block__change-source-wrapper"
        >
          <div
            class="googlesitekit-data-block__change googlesitekit-data-block__change--no-change"
          >
            <span
              class="googlesitekit-data-block__value"
            >
              0%
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-2-tablet"
    >
      <div
        class="googlesitekit-data-block overview-total-clicks googlesitekit-data-block--default"
        tabindex="-1"
      >
        <div
          class="googlesitekit-data-block__title-datapoint-wrapper"
        >
          <h3
            class=" googlesitekit-subheading-1 googlesitekit-data-block__title "
          >
            <span
              class="googlesitekit-data-block__title-inner"
            >
              Total Clicks
            </span>
          </h3>
          <div
            class="googlesitekit-data-block__datapoint"
          >
            123
          </div>
        </div>
        <div
          class="googlesitekit-data-block__change-source-wrapper"
        >
          <div
            class="googlesitekit-data-block__change googlesitekit-data-block__change--no-change"
          >
            <span
              class="googlesitekit-data-block__value"
            >
              0%
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mdc-layout-grid__cell mdc-layout-grid__cell--span-6-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div
        class="googlesitekit-analytics-cta"
      >
        <div
          class="googlesitekit-analytics-cta__preview-graphs"
        >
          <div
            class="googlesitekit-analytics-cta__preview-graph"
          >
            <h3
              class="googlesitekit-analytics-cta__preview-graph--title"
            >
              Traffic
            </h3>
            <div>
              <svg />
            </div>
          </div>
          <div
            class="googlesitekit-analytics-cta__preview-graph"
          >
            <h3
              class="googlesitekit-analytics-cta__preview-graph--title"
            >
              Unique visitors from Search
            </h3>
            <div>
              <svg />
            </div>
            <div
              class="googlesitekit-analytics-cta__preview-graph--icons"
            >
              <svg />
              <span
                class="googlesitekit-analytics-cta__preview-graph--bar"
              />
            </div>
          </div>
        </div>
        <div
          class="googlesitekit-analytics-cta__details"
        >
          <p
            class="googlesitekit-analytics-cta--description"
          >
            See how many people visit your site from Search and track how youâ€™re achieving your goals
          </p>
          <button
            class="mdc-button googlesitekit-button-icon--spinner googlesitekit-button-icon--spinner__after mdc-button--raised"
            target="_self"
          >
            <span
              class="mdc-button__label"
            >
              Set up Google Analytics
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
